### ISSUE TITLE
Unable to pip-sync on 1.10.0rc1

### ISSUE THREAD

**Issue by dfee** [jazzband/pip-tools#539]
Line 158 of [`pip-sync.py`](https://github.com/jazzband/pip-tools/blob/master/piptools/sync.py#L158) sorts the `to_install` set of InstallRequirements. However, actually attempting to pip-sync produces the following error:

```
/tmp/zebra via env [I] ➔ pip-sync
Traceback (most recent call last):
  File "/private/tmp/zebra/env/bin/pip-sync", line 11, in <module>
    load_entry_point('pip-tools', 'console_scripts', 'pip-sync')()
  File "/private/tmp/zebra/env/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/private/tmp/zebra/env/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/private/tmp/zebra/env/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/private/tmp/zebra/env/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/private/tmp/zebra/env/src/pip-tools/piptools/scripts/sync.py", line 73, in cli
    install_flags=install_flags))
  File "/private/tmp/zebra/env/src/pip-tools/piptools/sync.py", line 158, in sync
    for ireq in sorted(to_install):
TypeError: '<' not supported between instances of 'InstallRequirement' and 'InstallRequirement'
```

The example package, `zebra` consists of one file, `setup.py`. Here are the contents:
```
from setuptools import setup, find_packages

setup(
    name='zebra',
    version='0.1',
    packages=find_packages(),
    include_package_data=True,
    install_requires=[
        'pyramid',
        'pytest',
    ],
)
```

##### Environment Versions

1. Python version: 3.6.0
1. pip version:  9.0.1
1. pip-tools version: 1.10.0rc1

**Comment by jdufresne** [comment]
Duplicate of #569 which has been handled and released.

**Comment by vphilippon** [comment]
Fixed in 1.10.1
